linera-base/src/identifiers.rs

```diff
@@ -574,6 +584,92 @@ impl std::str::FromStr for Owner {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "AccountOwner")]
+enum SerializableAccountOwner {
+    User(Owner),
+    Application(ApplicationId),
+}
+
+impl Serialize for AccountOwner {
+    fn serialize<S: serde::ser::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            match self {
+                AccountOwner::Application(app_id) => SerializableAccountOwner::serialize(
+                    &SerializableAccountOwner::Application(*app_id),
+                    serializer,
+                ),
```

Comment:
```suggestion
                AccountOwner::Application(app_id) => SerializableAccountOwner::Application(*app_id).serialize(serializer),
```

Reply:
And then you could even deduplicate the `serialize(serializer)`, and move it out of the `match`.